control-plane/p4RuntimeArchHandler.h

```diff
@@ -414,11 +414,13 @@ struct Counterlike {
             BUG_CHECK(type->arguments->size() > *indexTypeParamIdx,
                       "%1%: expected at least %2% type arguments",
                       instance, *indexTypeParamIdx + 1);
-            auto typeArg = type->arguments->at(*indexTypeParamIdx);
-            // We ignore the return type on purpose, but the call is required to update p4RtTypeInfo
-            // if the index has a user-defined type.
-            TypeSpecConverter::convert(refMap, typeMap, typeArg, p4RtTypeInfo);
-            index_type_name = getTypeName(typeArg, typeMap);
+            const IR::Type* typeArg = type->arguments->at(*indexTypeParamIdx);
+            if (!typeArg->is<IR::Type_Dontcare>()) {
```

Comment:
Is this the right fix? index_type_name will be nullptr. 
Ideally you would use the inferred value for the type.
Perhaps the compiler should replace Type_Dontcare with the inferred value.

Reply:
Thank you for the comments.
Looking at `RemoveDontcareArgs` it is handling `IR::MethodCallExpression` but not `IR::Declaration_Instance`.
I will amend the PR by handling the need there.

Reply:
> Ideally you would use the inferred value for the type
Does this mean using the `TypeInference` class?
It is unclear to me if the `Type_Dontcare` should have been already inferred, or if the `TypeSpecConverter::convert()`  viz `TypeSpecConverter::preorder(const IR::Type* type)` should be using `TypeInference`.